exchange/bitswap/session.go

```diff
@@ -216,6 +223,11 @@ func (s *Session) run(ctx context.Context) {
 		case <-ctx.Done():
 			s.tick.Stop()
 			s.bs.removeSession(s)
+
+			cmgr := s.bs.network.ConnectionManager()
+			for _, p := range s.activePeersArr {
+				cmgr.UntagPeer(p, s.tag)
```

Comment:
We should consider adding a method for untagging all peers (although we'll have to consider the performance tradeoffs).